src/Utils/obiUtils.tsx

```diff
@@ -83,12 +83,45 @@ export function isSameActivity(activity1: BB.Activity, activity2: BB.Activity):
     return true
 }
 
+export async function lgMapFromLGFiles(lgFiles: File[] | null): Promise<Map<string, CLM.LGItem>> {
+    const lgMap: Map<string, CLM.LGItem> = new Map()
+    if (lgFiles) {
+        for (const lgFile of lgFiles) {
+            if (lgFile.name.endsWith('.lg')) {
+                const fileText = await Util.readFileAsync(lgFile)
+                CLM.ObiUtils.addToLGMap(fileText, lgMap)
+            }
+            else {
+                throw new Error(`Expecting .lg file.\n\n Given: ${lgFile.name}`)
+            }
+        }
+    }
+    return lgMap
+}
+
+export function substituteLG(text: string, lgMap: Map<string, CLM.LGItem> | null): string {
+    
+    // If not LG substitution just return the text
+    if (!lgMap || !text.startsWith('[') || !text.endsWith(']')) {
```

Comment:
These should be regularly-formed, so something like this would be safer :

```
const p = RegExp(/^\[\w+\]$/)
if (!lgMap || !p.test(text)) {
  return text
}
```